src/plugins/banner/public/plugin.ts

```diff
@@ -0,0 +1,93 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Any modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { CoreSetup, CoreStart, Plugin, PluginInitializerContext } from '../../../core/public';
+import { BannerPluginSetup, BannerPluginStart } from './types';
+import { bannerServiceInstance } from './services/banner_service_instance';
+import { GlobalBanner } from './components/global_banner';
+
+export class BannerPlugin implements Plugin<BannerPluginSetup, BannerPluginStart> {
+  constructor(_initializerContext: PluginInitializerContext) {}
+
+  public setup(core: CoreSetup): BannerPluginSetup {
+    // Setup banner with default values
+    bannerServiceInstance.setup({
+      text: 'This is an important announcement for all users.',
+      color: 'primary',
+      iconType: 'iInCircle',
+      isVisible: true,
+      useMarkdown: true,
+    });
+
+    return {};
+  }
+
+  public start(core: CoreStart): BannerPluginStart {
+    // Clear any stored banner height to prevent incorrect values from persisting
+    try {
+      localStorage.removeItem('opensearch-dashboards-banner-height');
+    } catch (e) {
+      // TODO
+      // Ignore localStorage errors for now
+    }
+
+    // Get the current banner config to determine initial height
+    const currentConfig = bannerServiceInstance.getCurrentConfig();
+
+    // Set initial height based on banner visibility
+    const initialHeight = currentConfig.isVisible ? '40px' : '0px';
```

Comment:
Nit: I'm seeing `40` and `0` get used a lot. Can we extract this in `common/constants.ts` to give more meaning to these values?